### DISCUSSION TITLE
Dynesty not initializing

### DISCUSSION THREAD

**Discussion by rfreire-400** [phoebe-project/phoebe2#869] [Questions]
I'm trying to use Dynesty to solve the inverse problem, but I've been having some issues running it. I'm unsure if the problem is with how I'm using it or if it's with Dynesty itself. I've added a zip file below with the example notebook and data. I'm using Phoebe 2.4.11 on Mac. I tried running this on a remote cluster, which gave the same results. 

<img width="927" alt="Screenshot 2024-05-13 at 2 55 31 PM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/14c9a53a-a00d-4fcd-a80d-c4113817c572">

<img width="927" alt="Screenshot 2024-05-13 at 2 56 01 PM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/51f15ec9-ab40-4935-9774-852fab7197a2">

<img width="927" alt="Screenshot 2024-05-13 at 2 56 21 PM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/341c5dc8-df90-4bd0-b703-60fa35c5bf8e">

[Dynesty problem example.zip](https://github.com/phoebe-project/phoebe2/files/15298362/Dynesty.problem.example.zip)

Thank you for your time

**Comment by Raindogjones** [comment]
Hi, I'm not that familiar with using dynesty but I'm wondering whether it could be the distributions you've set up. Perhaps it is the q distribution starting from an unphysical value of 0 that is causing an issue? Could you set it to say phoebe.uniform(0.001,1) and see if it throws the same error?

**Reply by rfreire-400** [reply]
Hi - I tried this and it gives the same error. 

<img width="927" alt="Screenshot 2024-05-15 at 9 56 33 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/9bdc551b-6075-48ee-87af-82e3196ba042">

<img width="927" alt="Screenshot 2024-05-15 at 9 58 54 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/e77bfa7f-d93f-4a7a-b656-de2f676497b2">

Dynesty doesn't have as much documentation as emcee so I'm unsure if I'm setting it up properly or not. My guess is that the solver is expecting a distribution labelled as 'init_from' (similar to how emcee works), but since dynesty doesn't have one, it's throwing an error.

<img width="927" alt="Screenshot 2024-05-15 at 9 59 48 AM" src="https://github.com/phoebe-project/phoebe2/assets/113457579/095f5724-8392-4e46-b7d6-8edde2c11cd8">

**Reply by Raindogjones** [reply]
Sorry, I'm not sure.  Unfortunately, the person who implemented the support for dynesty in phoebe is currently away. So, it might be a few days before you get a useful response.

**Reply by rfreire-400** [reply]
No worries, I can wait. Thanks anyways!

**Comment by rfreire-400** [comment]
Hi - I'm just checking in to see if there are any updates concerning this

**Reply by kecnry** [reply]
I will try to look into this this week, but it looks from the traceback that the bug itself is just in generating the warning message that "sampling with dataset-scaled can cause unintended consequences.  Consider using component-coupled or marginalizing over pblum".  If you are able to address that warning, then the bug should be inconsequential.  If you need to have dataset-scaled, then you can probably bypass this by passing `skip_checks=True` to `run_solver`.

**Reply by kecnry** [reply]
I _think_ the bug itself should be fixed by #874.  If you could test to see if that's the case, then we can get that in a bugfix sooner.  Otherwise I will try to reproduce the issue and test it against that patch sometime in the next week or so.  Thanks for your patience!

**Reply by rfreire-400** [reply]
I completely missed that warning somehow. It's hidden in the traceback so I didn't stop to read it. I've swapped "dataset-scaled" to "absolute" and it started to run, so I do think that was the issue. I'm trying to set up my environment so I don't use "dataset-scaled" and therefore get more accurate values, so this change is for the best.

Thanks a lot for your help!

**Reply by MichaelAbdul-Masih** [reply]
> I _think_ the bug itself should be fixed by #874. If you could test to see if that's the case, then we can get that in a bugfix sooner. Otherwise I will try to reproduce the issue and test it against that patch sometime in the next week or so. Thanks for your patience!

I've checked and can confirm that #874  does indeed fix this bug.

**Reply by kecnry** [reply]
Thanks @MichaelAbdul-Masih!  I'll merge that then and it will make it in the next bugfix release (2.4.14) sometime in the next two weeks.
